cube/cube.cpp

```diff
@@ -1362,32 +1232,20 @@ void Demo::init_vk() {
         fprintf(stderr, "Selected GPU %d: %s, type: %s\n", gpu_number, physicalDeviceProperties.deviceName.data(),
                 to_string(physicalDeviceProperties.deviceType).c_str());
     }
-    physical_devices.reset();
 
     /* Look for device extensions */
-    uint32_t device_extension_count = 0;
     vk::Bool32 swapchainExtFound = VK_FALSE;
-    enabled_extension_count = 0;
-    memset(extension_names, 0, sizeof(extension_names));
 
-    result =
-        gpu.enumerateDeviceExtensionProperties(nullptr, &device_extension_count, static_cast<vk::ExtensionProperties *>(nullptr));
-    VERIFY(result == vk::Result::eSuccess);
-
-    if (device_extension_count > 0) {
-        std::unique_ptr<vk::ExtensionProperties[]> device_extensions(new vk::ExtensionProperties[device_extension_count]);
-        result = gpu.enumerateDeviceExtensionProperties(nullptr, &device_extension_count, device_extensions.get());
-        VERIFY(result == vk::Result::eSuccess);
+    auto device_extension_return = gpu.enumerateDeviceExtensionProperties();
+    VERIFY(device_extension_return.result == vk::Result::eSuccess);
 
-        for (uint32_t i = 0; i < device_extension_count; i++) {
-            if (!strcmp(VK_KHR_SWAPCHAIN_EXTENSION_NAME, device_extensions[i].extensionName)) {
-                swapchainExtFound = 1;
-                extension_names[enabled_extension_count++] = VK_KHR_SWAPCHAIN_EXTENSION_NAME;
-            }
-            if (!strcmp("VK_KHR_portability_subset", device_extensions[i].extensionName)) {
-                extension_names[enabled_extension_count++] = "VK_KHR_portability_subset";
-            }
-            assert(enabled_extension_count < 64);
+    for (const auto &extension : device_extension_return.value) {
+        if (!strcmp(VK_KHR_SWAPCHAIN_EXTENSION_NAME, extension.extensionName)) {
+            swapchainExtFound = 1;
+            enabled_device_extensions.push_back(VK_KHR_SWAPCHAIN_EXTENSION_NAME);
+        }
+        if (!strcmp("VK_KHR_portability_subset", extension.extensionName)) {
```

Comment:
Could be `else if`

Reply:
agreed. Added in another place where extensions are being checked.